### PR TITLE
Feature/backend/default device

### PR DESCRIPTION
**Pull request by gdchamal**
Create a default device upon user registration

### CODE REVIEW

File: src/backend/components/py.pi/caliopen_pi/features/device.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+            ua = parse_ua(device.user_agent)
+            browser_version = ua.browser.version_string.lower()
+            features.update({'browser_family': ua.browser.family.lower(),
+                             'browser_version': browser_version,
+                             'os_family': ua.os.family.lower(),
+                             'os_version': ua.os.version_string.lower(),
+                             'device_family': ua.device.family.lower(),
+                             'device_type': self._get_device_type(ua)})
+        # XXX processing IP address to detect some informations
+        features.update(self._process_ip_address(device.ip_address))
```
[the remark this answers is not in the document]
Comment (gdchamal) [reply src/backend/components/py.pi/caliopen_pi/features/device.py]: ip_creation, I updated parameter and processing on DeviceQualifier

File: src/backend/main/py.main/caliopen_main/device/core.py [hunk 2]
```diff
@@ -39,4 +40,14 @@ def create(cls, user, device, **related):
 
         core = super(Device, cls).create(user, **attrs)
         log.debug('Created device %s' % core.device_id)
```
[the remark this answers is not in the document]
Comment (gdchamal) [reply src/backend/main/py.main/caliopen_main/device/core.py]: yes it's a related. It's updated

File: src/backend/main/py.main/caliopen_main/device/parameters.py [hunk 3]
```diff
@@ -4,13 +4,13 @@
...
 from schematics.transforms import blacklist
-from schematics.types import DateTimeType, StringType, UUIDType, LongType
+from schematics.types import DateTimeType, StringType, UUIDType
 from schematics.types.compound import ListType, ModelType, DictType
 
 from caliopen_main.pi.parameters import PIParameter
 import caliopen_storage.helpers.json as helpers
 
-DEVICE_TYPES = ['unknow', 'desktop', 'laptop', 'smartphone', 'tablet']
+DEVICE_TYPES = ['other', 'desktop', 'laptop', 'smartphone', 'tablet']
```
[the remark this answers is not in the document]
Comment (gdchamal) [reply src/backend/main/py.main/caliopen_main/device/parameters.py]: done

File: src/backend/main/py.main/caliopen_main/user/parameters/device.py [hunk 4]
```diff
@@ -20,22 +20,15 @@ class DeviceLocation(Model):
...
 class NewDevice(Model):
     """Structure to create a new user device."""
 
     name = StringType(required=True)
     type = StringType(required=True, choices=DEVICE_TYPES, default='unknow')
 
     locations = ListType(ModelType(DeviceLocation), default=lambda: [])
-    informations = ModelType(DeviceInformation)
+    user_agent = StringType()
+    ip_address = StringType()
```
[the remark this answers is not in the document]
Comment (gdchamal) [reply src/backend/main/py.main/caliopen_main/user/parameters/device.py]: fixed

File: src/backend/defs/rest-api/objects/Device.yaml [hunk 5]
```diff
@@ -4,24 +4,33 @@ properties:
...
+  date_insert:
+    type: string
+    format: date-time
+  date_revoked:
+    type: string
+    format: date-time
+  ip_creation:
+    type: string
+  name:
     type: string
```
[the remark this answers is not in the document]
Comment (gdchamal) [reply src/backend/defs/rest-api/objects/Device.yaml]: fixed

File: src/backend/main/py.main/caliopen_main/device/core.py [hunk 6]
```diff
@@ -25,12 +24,14 @@ class Device(BaseUserCore, MixinCoreNested):
         'locations': ModelDeviceLocation,
```
[the remark this answers is not in the document]
Comment (gdchamal) [reply src/backend/main/py.main/caliopen_main/device/core.py]: yes, fixed

**Comment by gdchamal** [comment]
Creating a device without an ECDSA key is not permitted. Enforced in swagger definition.

- I can't reproduce case where device is not created on POST /users.
- Can't generate a device using cli, as private ecdsa key will not exist

**Comment by sapiens-sapide** [comment]
I succeeded to create a user with a default device, but the `ip_creation` and `user_agent`  properties are empty in cassa.

**Comment by sapiens-sapide** [comment]
LGTM

**Pull request merged**